appserver/tests/payara-samples/samples/rest-management/src/test/java/fish/payara/samples/rest/management/CreateInstanceTest.java

```diff
@@ -37,42 +37,57 @@
  * only if the new code is made subject to such option by the copyright
  * holder.
  */
-package fish.payara.samples.rest.management.extension;
 
-import java.lang.annotation.Annotation;
-import java.net.URL;
+package fish.payara.samples.rest.management;
 
-import org.jboss.arquillian.container.test.impl.enricher.resource.ContainerURLResourceProvider;
-import org.jboss.arquillian.core.api.annotation.Observes;
-import org.jboss.arquillian.test.api.ArquillianResource;
-import org.jboss.arquillian.test.spi.event.suite.AfterClass;
-import org.jboss.arquillian.test.spi.event.suite.BeforeClass;
+import fish.payara.samples.CliCommands;
+import fish.payara.samples.NotMicroCompatible;
+import fish.payara.samples.PayaraArquillianTestRunner;
+import fish.payara.samples.PayaraTestShrinkWrap;
+import static fish.payara.samples.rest.management.RestManagementTest.INSTANCE_NAME;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
 
-public class TemporaryInstanceProvider extends ContainerURLResourceProvider {
-
-    private static TemporaryInstance instance;
-
-    @Override
-    public boolean canProvide(Class<?> type) {
-        return type.isAssignableFrom(TemporaryInstance.class);
+/**
+ *
+ * @author lprimak
+ */
+@RunWith(PayaraArquillianTestRunner.class)
+@NotMicroCompatible
+public class CreateInstanceTest {
+    @Deployment
+    public static WebArchive deploy() {
+        return PayaraTestShrinkWrap.getWebArchive();
     }
 
-    @Override
-    public Object lookup(ArquillianResource resource, Annotation... qualifiers) {
-        if (!instance.isInitialized()) {
-            instance.initialise((URL) super.lookup(resource, qualifiers));
+    @Test
+    public void createInstance() {
+        List<String> output = new ArrayList<>();
+        CliCommands.payaraGlassFish(Arrays.asList("list-instances"), output);
+        List<String> instances = output.stream().map(this::firstWord).collect(Collectors.toList());
+        if (!instances.contains(INSTANCE_NAME)) {
+            output.clear();
+            CliCommands.payaraGlassFish(Arrays.asList("list-nodes"), output);
+            String node = output.stream().map(this::firstWord).findFirst().get();
+            CliCommands.payaraGlassFish("create-instance", "--node", node, INSTANCE_NAME, "--terse");
         }
-        return instance;
     }
 
-    public void beforeClassEvent(@Observes BeforeClass event) {
-        instance = new TemporaryInstance();
-    }
-
-    public void afterClassEvent(@Observes AfterClass event) {
-        if (instance.isCreated()) {
-            instance.destroy();
+    String firstWord(String line) {
+        if (line == null) {
+            return line;
+        }
+        String[] words = line.split(" ");
+        if (words.length > 0) {
```

Comment:
Shouldn't this be > 1?
It's not _wrong_ per se, but splitting a line with no spaces will return an array with one element, so surely this always passes as long as line isn't null?

Reply:
you are right :) useless code and will be removed